makefiles/suit.v4.inc.mk

```diff
@@ -16,18 +16,38 @@ export CFLAGS += -DSOCK_URLPATH_MAXLEN=128
 SUIT_VENDOR ?= "riot-os.org"
 SUIT_SEQNR ?= $(APP_VER)
 SUIT_DEVICE_ID ?= $(BOARD)
-SUIT_KEY ?= secret.key
-SUIT_PUB ?= public.key
-SUIT_PUB_HDR ?= public_key.h
+
+#
+# SUIT encryption keys
+#
+ifeq (1, $(RIOT_CI_BUILD))
+  SUIT_KEY_DIR ?= $(BINDIR)
+else
+  SUIT_KEY_DIR ?= $(RIOTBASE)/keys
+endif
+
+SUIT_KEY ?= $(SUIT_KEY_DIR)/secret.key
+SUIT_PUB ?= $(SUIT_KEY_DIR)/public.key
+
+SUIT_PUB_HDR = $(BINDIR)/riotbuild/public_key.h
+CFLAGS += -I$(dir $(SUIT_PUB_HDR))
+BUILDDEPS += $(SUIT_PUB_HDR)
 
 $(SUIT_KEY) $(SUIT_PUB):
-	@$(RIOTBASE)/dist/tools/suit_v4/gen_key.py
+	@echo suit: generating key pair in $(SUIT_KEY_DIR)
+	@mkdir -p $(SUIT_KEY_DIR)
+	@$(RIOTBASE)/dist/tools/suit_v4/gen_key.py $(SUIT_KEY) $(SUIT_PUB)
 
 $(SUIT_PUB_HDR): $(SUIT_PUB)
-	@xxd -i $(SUIT_PUB) > $@
+	@mkdir -p $(dir $(SUIT_PUB_HDR))
+	@cd $(dir $(SUIT_PUB)) && xxd -i $(notdir $(SUIT_PUB)) > $@
```

Comment:
`$(dir $(SUIT_PUB))` contains the same value as `$(SUIT_KEY_DIR)`. Maybe just use the later ?

Reply:
No, SUIT_PUB can be overridden, it just defaults to "$(SUIT_KEY_DIR)/filename". E.g., ```SUIT_PUB="~/keys/production.pub" make``` could fail if re-using the variable.

Reply:
Ok, I see. Makes sense.